ztools/ztls/handshake_messages.go

```diff
@@ -99,6 +101,17 @@ func (m *clientHelloMsg) marshal() []byte {
 		extensionsLength += 1
 		numExtensions++
 	}
+	if len(m.alpnProtocols) > 0 {
+		extensionsLength += 2
+		for _, s := range m.alpnProtocols {
+			if l := len(s); l == 0 || l > 255 {
+				panic("invalid ALPN protocol")
```

Comment:
I think it is appropriate to panic() here. This panic is not triggered by server, but by client, when he's doing something very wrong: you'd want to know about misconfiguration.